[FIX] l10n_es_aeat: specify cryptography version

yajo opened the pull request:
This module requires the specified cryptography version. Make that explicit in its manifest, now that thanks to https://github.com/odoo/odoo/pull/25549 we can specify the package version.

@moduon MT-1067

yajo committed:
[FIX] l10n_es_aeat: specify cryptography version

This module requires the specified cryptography version. Make that explicit in its manifest, now that thanks to https://github.com/odoo/odoo/pull/25549 we can specify the package version.

@moduon MT-1067

pedrobaeza commented on requirements.txt:
```diff
@@ -1,6 +1,7 @@
 # generated from manifests external_dependencies
 chardet
 cryptography
```
Duplicado

yajo replied:
Esto lo añade el hook de pre-commit. Debe ser que hay otro módulo en el repo que pide esta dependencia sin versión. Tampoco es que sea un problema.

pedrobaeza replied:
Podrías mirar cuál es y unificarlo?

pedrobaeza commented:
De todas formas, el enlace a Odoo solo indica la característica de que se puede especificar una versión en el manifiesto, no la razón por la que aquí se indica `> 3`. ¿Cuál es esa razón?

yajo commented:
Es un requisito actual del módulo: https://github.com/OCA/l10n-spain/blob/279167b0be43ba1e3ebc043ef62a687290ca6cb9/l10n_es_aeat/wizard/aeat_certificate_password.py#L29-L32

Supongo que esa comprobación viene de otras versiones donde no se podía hacer esto. Por ahora creo que es mejor no quitarla, ya que puede estar ya instalado en algún sitio con una versión anterior de cryptography, pero igual al migrarlo a v16 ya se puede quitar ese warning.

pedrobaeza commented:
Vale, gracias por explicarlo.

Añade por favor un comentario `# FIXME: To be removed in v16 as specified in the manifest` encima de esa comprobación y en otra que se hace luego más tarde (https://github.com/OCA/l10n-spain/commit/373198ddfa6bfe4bc5bf89a308d5cbbd703206b8), y así no se nos olvida.

¿Has encontrado el otro sitio donde aparece?

yajo commented:
> ¿Has encontrado el otro sitio donde aparece?

Hay bastantes módulos, pero todos acaban dependiendo de este. ¿Tiene sentido tener que repetir la dependencia python si ya la incluye un módulo del que depende? Porque tengo 3 opciones:

1. Quitar la dependencia de sus manifests.
2. Aumentarla de versión.
3. Dejarlo como está (ya que realmente no hay conflicto).

¿Qué prefieres?

pedrobaeza commented:
Esto también es una cuestión histórica: antes el certificado se ponía en el SII, y luego se trasladó al módulo principal, pero se debió quedar esto como leftover. Supongo que el resto de los que lo incluyen son los de TicketBAI y Batuz, que habrán pasado por una situación similar. Creo que la mejor opción es la 1 (limpiarlo), y así evitar estos enmarañamientos históricos.

yajo committed:
[FIX] l10n_es_*: remove pyOpenSSL and cryptography dependencies

- pyOpenSSL was replaced by cryptography and is no longer used.
- cryptography has a specific version dependency specified in `l10n_es_aeat`, which in turn is dependency of all these modules. So there's no real need to indicate that dependency in these files.

@moduon MT-1067

yajo commented:
OK pues ya lo tienes.

pedrobaeza reviewed (approved):
Gracias!

pedrobaeza commented:
/ocabot merge patch

pedrobaeza commented:
@yajo tenemos un problema con esto debido al requirements de Odoo que dice que no debe ser esa versión:

https://github.com/odoo/odoo/blob/55d2da4dd28975e78ace2a6beeae4168d5692f09/requirements.txt#L3

@etobella recuerdas por qué pusiste que tenía que ser >= 3 ?

etobella commented:
Se hizo con 3 por que en el momento de la introducción de la dependencia odoo no la tenia incluida

https://github.com/OCA/l10n-spain/pull/1705

En ese momento no estaba fijada la versión de cryptography en odoo, ya que no estaba en los requerimientos. Por eso se usó la nueva. Si fuera necesario, se podría revisar a ver como cambiar a usar pyopenssl.

De hecho, la limitación viene dada por pyopenssl==19.0 que no se introdujo en odoo hasta v15.0, por lo que en ese momento se decidió usar la versión más actualizada disponible.

De hecho, el cambio de pyopenssl se intpodriarodujo en master en este PR

https://github.com/odoo/odoo/pull/61853

Y el de cryptography aquí (hace unos pocos días)

https://github.com/odoo/odoo/pull/99829

De todas formas, se debe tener en cuenta que hacer este cambio hará necesario revisar la libreria xmlsig, ya que también espera una versión más actualizada de cryptography para trabajar ....

pedrobaeza commented:
¿Se podría conseguir lo mismo con la versión de `cryptography` fijada en Odoo?

etobella commented:
Se tiene que probar, pero seguramente se nrcesitaran bastantes cambios.

Puedo mirarlo, pero no prometo nada 😢
